sql/hive/src/main/scala/org/apache/spark/sql/hive/execution/HiveFileFormat.scala

```diff
@@ -142,7 +142,8 @@ class HiveOutputWriter(
 
   override def write(row: InternalRow): Unit = {
     var i = 0
-    while (i < fieldOIs.length) {
+    val fieldOIsLength = fieldOIs.length
+    while (i < fieldOIsLength) {
```

Comment:
`fieldOIs` is an array, and accessing its length should be as fast as accessing any local variable. I can't imagine it's making this much difference compared to all the other operations here. Are you sure?